packages/jest-environment-puppeteer/src/PuppeteerEnvironment.js

```diff
@@ -47,7 +47,17 @@ class PuppeteerEnvironment extends NodeEnvironment {
           : undefined,
       browserWSEndpoint: wsEndpoint,
     })
-    this.global.page = await this.global.browser.newPage()
+
+    if(config.launch.browserContext === 'incognito') {
+      this.global.context = await this.global.browser.createIncognitoBrowserContext()
+      // Create a new page in a pristine context.
+      this.global.page = await this.global.context.newPage()
+    } else if(config.launch.browserContext === 'default') {
```

Comment:
Are you OK with explicitely checking this or you would rather just have `else` ? I would like to avoid allowing users to specify anything else than `default` or `incognito`